server/bleep/src/agent.rs

```diff
@@ -57,7 +57,8 @@ pub struct Agent {
     pub thread_id: uuid::Uuid,
     pub query_id: uuid::Uuid,
 
-    pub model: model::AnswerModel,
+    pub answer_model: model::LLMModel,
+    pub agent_model: model::LLMModel,
```

Comment:
i think its confusing that `Agent` has, within itself, an `agent_model` field, can we call this something else? such as `decsicion_model` or `reasoning_model`?

Reply:
I think `agent_model` is fine.
The abstraction is:  the user queries to get an `answer` and we have an `answer_model` used in the tool `answer`, to get the the correct context to answer the query we start an `agent` which has its own `agent_model`

Third opinion @ggordonhall ?